packages/app/src/sandbox/compile.ts

```diff
@@ -667,9 +668,19 @@ async function compile(opts: CompileOptions) {
 
       await manager.preset.preEvaluate(manager, updatedModules);
 
-      if (!manager.webpackHMR) {
-        const htmlEntries = templateDefinition.getHTMLEntries(configurations);
-        const htmlModulePath = htmlEntries.find(p => Boolean(modules[p]));
+      // HTML Hydration and HMR
+      // If the user code handles hot module reload, we don't modify html dynamically (users can refresh the page manually in this case)
+      if (!manager?.webpackHMR) {
+        let pathname = document.location.pathname;
+        if (!pathname.endsWith('.html')) {
+          pathname = '/index.html';
+        }
+        const htmlEntries = Object.keys(modules).filter(k =>
```

Comment:
Changed this logic as this ensures we can serve any html file in sandpack (even multi page ones)